Fix Sonar issues Cli and CliAutoComplete

McGiverGim opened the pull request:
Cleaning of major part of Sonar issues in CLI and the CLIAutoComplete parts.

haslinghuis commented on src/js/CliAutoComplete.js:
```diff
@@ -193,55 +203,57 @@ CliAutoComplete.builderParseLine = function(line) {
+        const res = [];
 
-        if ((minChars !== false && term.length >= minChars) || self.forceOpen || self.isOpen()) {
+        if ((minChars!== false && term.length >= minChars) || self.forceOpen || self.isOpen()) {
```
```suggestion
        if ((minChars !== false && term.length >= minChars) || self.forceOpen || self.isOpen()) {
```
Is this intentional?

McGiverGim replied:
Fixed!

haslinghuis commented on src/js/CliAutoComplete.js:
```diff
@@ -193,55 +203,57 @@ CliAutoComplete.builderParseLine = function(line) {
-                if (matchPrefix && v.startsWith(term) || !matchPrefix && v.indexOf(term) !== -1) {
+            for (let i = 0; i < array.length; i++) {
+                const v = array[i].toLowerCase();
+                if (matchPrefix && v.startsWith(term) || !matchPrefix && v.indexOf(term)!== -1) {
```
```suggestion
                if (matchPrefix && v.startsWith(term) || !matchPrefix && v.indexOf(term) !== -1) {
```
Same.

McGiverGim replied:
Fixed!

McGiverGim commented:
Fixed as per review (thanks @haslinghuis !)

McGiverGim commented:
Rebased against master, to get the real number of code smells...

haslinghuis reviewed (approved):
Thanks. LGTM!

mikeller commented:
Weirdly Travis tests are failing for this change.

McGiverGim commented:
I have moved the obtention of some GUI elements to a method that the tests do not call and for this reason they are failing. I will look at this later, I have not experience with this Karma thing...

McGiverGim commented:
Fixed, but I'm not an expert in this Karma thing so maybe I have made some mistake:
- I have modified the tests to have all the elements that I reordered, so the tests can be executed.
- I have added the `spec` reporter, because I was not able to see the result of the tests. Maybe this is wrong and I need to remove it? I have not removed the old `tfs` because maybe is used at some place unknown to me...
- Fixed some Sonar issues in this tests files too...
- Rebased against master...

mikeller commented on package.json:
```diff
@@ -55,8 +55,7 @@
     "jquery": "^3.5.1",
-    "jquery-textcomplete": "^1.8.5",
-    "jquery-touchswipe": "^1.6.19",
+    "jquery-textcomplete": "^1.8.5",    "jquery-touchswipe": "^1.6.19",
```
Not sure what happened here.

McGiverGim replied:
Me neither... it's clear that I'm not at my best moment 🤸‍♂️

mikeller reviewed (approved):
Apart from the weirdness in `package.json` this looks good to me.

McGiverGim commented:
Now we have this output in the tests:

![image](https://user-images.githubusercontent.com/2673520/96446238-6e58f600-1211-11eb-874b-5763283b1167.png)

I don't know if is ok for our Jenkins output or we do prefer to remove it.

haslinghuis reviewed (approved):
🤗
